gennav/utils/graph.py

```diff
@@ -0,0 +1,39 @@
+from collections import defaultdict
+
+from gennav.utils import RobotState
+from gennav.utils.common import Node
+
+
+class Graph:
+    """Graph class used for graph based algorithms.
+    """
+
+    def __init__(self):
+        self.nodes = set()
+        self.edges = defaultdict(list)
+        self.distances = {}
+
+    def add_node(self, node=Node(state=RobotState())):
```

Comment:
Is it neccesarry to have default value for the node argument?

Reply:
No, it's not necessary to include `node=Node(state=RobotState())`, just `node=Node()` is fine. If I don't include that, then even a Point could be added to the graph. So, this just puts a constraint on the user if he just decides to use the Graph class directly (for whatever reason). 
This imagination might be a little far off, so if you want, I can remove the default value.